operator/pkg/validate/validate.go

```diff
@@ -200,6 +201,33 @@ func validateRevision(_ util.Path, val interface{}) util.Errors {
 	return nil
 }
 
+func validateRevisionFromIOPYAML(iopYAML string) error {
+	var specTree = make(map[string]interface{})
+	spec, err := tpath.GetConfigSubtree(iopYAML, "spec")
+	if err != nil {
+		return err
+	}
+	err = yaml.Unmarshal([]byte(spec), &specTree)
+	if err != nil {
+		return fmt.Errorf("error unmarshalling spec overlay yaml into untype tree %v", err)
+	}
+	rev := specTree["revision"]
+	// Skip if revision is not specified in any of the profile
+	if util.IsEmptyString(rev) {
+		return nil
+	}
+	revision, ok := rev.(string)
+	// fail early if revision is not a string eg: revision: 18, revision: 1.8, revision: 1.8.0
+	if !ok {
+		return fmt.Errorf("invalid revision specified: %v", rev.(float64))
```

Comment:
This line is no good because it causes istioctl to backtrace if supplied 

```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
spec:
  profile: default
  revision: false
```

Reply:
Fixed. Thanks.